src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1127,12 +1127,23 @@ private static bool IsUnboundGeneric(MethodSymbol method)
             return method.IsGenericMethod && method.ConstructedFrom() == method;
         }
 
+        // Arbitrary limit on the number of candidate methods for
+        // binding argument types for error recovery. If there are
+        // more candidates than this, no candidates are considered.
+        private const int MaxCandidatesForErrorRecovery = 4;
```

Comment:
>private const int MaxCandidatesForErrorRecovery = 4; [](start = 8, length = 52)

It feels like the number is too low, perhaps 20, or 40 as you suggested offline?